google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStorageFileSystem.java

```diff
@@ -335,4 +335,45 @@ public String toString() {
       throw new AssertionError(e);
     }
   }
+
+  private static final class ConfigTuple {
```

Comment:
Add javadoc on why this ConfigTuple is needed.

Reply:
Added a javadoc and renamed to `ProviderCacheKey` to be a bit less generic